src/main/java/com/google/opensesame/user/UserEntity.java

```diff
@@ -0,0 +1,123 @@
+package com.google.opensesame.user;
+
+import com.googlecode.objectify.annotation.Entity;
+import com.googlecode.objectify.annotation.Id;
+import com.googlecode.objectify.annotation.Index;
+import com.googlecode.objectify.annotation.OnSave;
+import java.util.ArrayList;
+
+/**
+ * The User Entity is designed to directly interface with datastore using Objectify. It holds the
+ * neccesary information to generate a UserObject.
+ */
+@Entity
+public class UserEntity {
+  @Id public String userId;
+  @Index Boolean isMentor;
+  public String email;
+  public String gitHubId;
+  /** The interests a user indicated during signup */
+  public ArrayList<String> interestTags;
+  /** Datastore IDs of projects the given user has decided to mentor for. */
+  public ArrayList<String> projectIds = new ArrayList<String>();
+  /** Datastore IDs of mentees that have expressed interest in being mentored by this user. */
+  public ArrayList<String> menteeIds = new ArrayList<String>();
+
+  public UserEntity() {}
+
+  /**
+   * Create a User entity with information from a UserData Instance
+   *
+   * @param user
+   */
+  public UserEntity(UserData userData) {
+    gitHubId = userData.getGitHubID();
+    interestTags = userData.getTags();
+    email = userData.getEmail();
+    userId = userData.getUserID();
+    projectIds = userData.getProjectIDs();
+    menteeIds = userData.getMenteeIDs();
+  }
+
+  // TODO: Cut down the number of constructors.
```

Comment:
@Obinnabii Is there a reason why we couldn't just pass in null or an empty list for the `projectIds` and `menteeIds` to avoid having these extra constructors? We could handle null however we wanted (including creating empty lists when the inputted value is null).

Reply:
my main issue with passing in null values is that it is not good practice. The standard with Java is to avoid using ```null``` as much as possible. It would also make the documentation a bit more confusing. This way, you always know what you have to put in.

__Possible Solutions__
1. Forgo a constructor entirely. The class can be instantiated without a constructor and all the fields can be set because they are public. 
_con_
- The main shortcoming of this method is that there are some fields we want to ensure always gets set. My current solution works so that you always instantiate a ```UserEntity``` object with the required fields and any optional fields are set as empty lists. 
2. A builder. This might work. 
_con_
- I am reluctant to add more classes to the 2 that we have now. and It would also make allow the compulsory fields to be left unset.

3. Leave the constructors as we have them. 
_con_
- we currently have 4 constructors.

4. Create the ```UserEntity``` Object without the optional fields and set those fields after construction.
_con_
- It seems hacky and construction of objects could take multiple lines depending on the situation.